### PR TITLE
[GCC11] Add missing #include for std::size_t definition

### PR DESCRIPTION
**Pull request by iarspider**
Error log: https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc11/CMSSW_12_1_X_2021-08-09-1100/HeterogeneousCore/CUDAUtilities
Error message: 
```
In file included from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/ada76bbef987ba1823d9aebd3a0b2f87/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-08-09-1100/src/HeterogeneousCore/CUDAUtilities/src/deviceAllocatorStatus.cc:1:
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/ada76bbef987ba1823d9aebd3a0b2f87/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-08-09-1100/src/HeterogeneousCore/CUDAUtilities/interface/deviceAllocatorStatus.h:10:9: error: 'size_t' does not name a type
    10 |         size_t free;
      |         ^~~~~~
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/ada76bbef987ba1823d9aebd3a0b2f87/opt/cmssw/slc7_amd64_gcc11/cms/cmssw/CMSSW_12_1_X_2021-08-09-1100/src/HeterogeneousCore/CUDAUtilities/interface/deviceAllocatorStatus.h:5:1: note: 'size_t' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
    4 | #include <map>
  +++ |+#include <cstddef>
    5 | 
```

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by iarspider** [comment]
please test for slc7_amd64_gcc11

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
gcc11 build errors are fix ( https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9ee046/17690/build-logs/HeterogeneousCore/CUDAUtilities/log.html ) though we have still have few more in tests which can be fixed in a different PR

**Comment by smuzaffar** [comment]
@fwyzard any objection on this change? If this is the correct fix then it should fix many build/link errors for gcc11 IBs

**Comment by fwyzard** [comment]
no objections to the change itself, but why is it needed ?
isn't `std::size_t` the same type as `size_t` ?

also, if we make the change we need to add an explicit comment about it, as per the file's license

File: HeterogeneousCore/CUDAUtilities/interface/deviceAllocatorStatus.h [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 #include <map>
 
 namespace cms {
   namespace cuda {
     namespace allocator {
       struct TotalBytes {
-        size_t free;
-        size_t live;
-        size_t liveRequested;  // CMS: monitor also requested amount
+        std::size_t free;
```
Comment (fwyzard) [inline HeterogeneousCore/CUDAUtilities/interface/deviceAllocatorStatus.h:11]:
```suggestion
        // CMS: add explicit std namespace
        std::size_t free;
```

**Comment by iarspider** [comment]
`size_t` and `std::size_t` are the same by definition:
```
#include <stddef.h>
namespace std 
{
  using ::ptrdiff_t;
  using ::size_t;
}
```

When writing in C++, one can either include `stddef.h` and use regular `size_t`, or include `cstddef` and use `std::size_t`. Of course, when writing in pure C, the latter is not an option.

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
+heterogeneous

**Comment by perrotta** [comment]
+1
- Technical

**Pull request merged**